utilities/transactions/write_unprepared_txn.cc

```diff
@@ -46,7 +46,408 @@ SequenceNumber WriteUnpreparedTxnReadCallback::MaxUnpreparedSequenceNumber() {
 WriteUnpreparedTxn::WriteUnpreparedTxn(WriteUnpreparedTxnDB* txn_db,
                                        const WriteOptions& write_options,
                                        const TransactionOptions& txn_options)
-    : WritePreparedTxn(txn_db, write_options, txn_options), wupt_db_(txn_db) {}
+    : WritePreparedTxn(txn_db, write_options, txn_options), wupt_db_(txn_db) {
+  max_write_batch_size_ = txn_options.max_write_batch_size;
+  write_batch_.SetMaxBytes(0);
```

Comment:
I'll add a comment. The reason is that we don't want to limit the size of the write batch, since we'll be size checking it ourselves.

The alternative that I was considering is to catch the size limit error and retry appending to batch, but I chose against that because checking first seemed more efficient.